configure.py

```diff
@@ -163,6 +164,41 @@ def flag_supported(flag, compiler):
     return try_compile(flags=['-Werror'] + split, compiler=compiler)
 
 
+class OptimizationLevel:
+    warned = False
+
+    @classmethod
+    def _warn(cls, safe_level) -> None:
+        if cls.warned:
+            # only prints this warning for a single time
+            return
+        print('\033[91mWARN\033[00m: '
+              f'Changing optimization level to "-O{safe_level}" '
+              'due to https://github.com/llvm/llvm-project/issues/62842. '
+              'Please note -O0 is so slow that some tests might fail.')
```

Comment:
This isn't true. @avikivity mistook a failure of the CI infrastructure for a problem with O0.

Reply:
makes sense to me . i'm also observing test timeouts recently after the `-O0` change was reverted.